python-package/lightgbm/sklearn.py

```diff
@@ -454,6 +454,30 @@ def __call__(
     """
 
 
+def _get_meta_data(
```

Comment:
While moving this to a top-level function, can we use a more expressive name? E.g.

```suggestion
def _extract_evaluation_meta_data(
```

Reply:
sure! added in https://github.com/microsoft/LightGBM/pull/6470/commits/7e48939b50e41a27c5e81eaa055bce8c11dd7b9a